### PR TITLE
Allow to save local artifact URI as relative

### PR DESCRIPTION
**Pull request by schipiga**
Provides ability for experiment and run to save artifact URI as relative path optionally.
It will be helpful to provide portability of mlruns, to share them between several machines, to save to git repo, to capture in docker image, etc.

### CODE REVIEW

File: mlflow/store/file_store.py [hunk 1]
```diff
@@ -173,6 +177,8 @@ def _get_experiment(self, experiment_id, view_type=ViewType.ALL):
             raise MlflowException("Could not find experiment with ID %s" % experiment_id,
                                   databricks_pb2.RESOURCE_DOES_NOT_EXIST)
         meta = read_yaml(experiment_dir, FileStore.META_DATA_FILE_NAME)
+        if is_fs_path(self.artifact_root_uri):
+            meta['artifact_location'] = build_path(self.artifact_root_uri, meta['artifact_location'])
```
Comment (tomasatdatabricks) [inline mlflow/store/file_store.py]: Why do you need this?

File: mlflow/store/file_store.py [hunk 2]
```diff
@@ -140,15 +140,18 @@ def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
             rsl += self._get_deleted_experiments(full_path=False)
         return [self._get_experiment(exp_id, view_type) for exp_id in rsl]
 
-    def _create_experiment_with_id(self, name, experiment_id, artifact_uri):
+    def _create_experiment_with_id(self, name, experiment_id, artifact_uri, artifact_relative):
```
Comment (tomasatdatabricks) [inline mlflow/store/file_store.py]: Can you not achieve the same by setting store.artifact_root_uri to store. root_directory?

File: mlflow/store/file_store.py [hunk 3]
```diff
@@ -282,6 +289,8 @@ def create_run(self, experiment_id, user_id, run_name, source_type,
                     databricks_pb2.INVALID_STATE)
         run_uuid = uuid.uuid4().hex
         artifact_uri = self._get_artifact_dir(experiment_id, run_uuid)
+        if artifact_relative and is_fs_path(self.artifact_root_uri):
```
Comment (tomasatdatabricks) [inline mlflow/store/file_store.py]: Isn't the artifact_relative a property of the experiment?

**Review by tomasatdatabricks — COMMENTED** [review]
I am not sure I understand purpose of this PR. Can you please elaborate?

**Comment by juntai-zheng** [comment]
Closing this PR due to inactivity. Feel free to open another or reopen if you want to work on this more!

**Pull request closed without merging**